app/components/editable_champ/editable_champ_component.rb

```diff
@@ -1,5 +1,48 @@
 class EditableChamp::EditableChampComponent < ApplicationComponent
-  def initialize(form:, champ:)
-    @form, @champ = form, champ
+  def initialize(form:, champ:, seen_at: nil)
+    @form, @champ, @seen_at = form, champ, seen_at
+  end
+
+  private
+
+  def has_label?(champ)
+    types_without_label = [TypeDeChamp.type_champs.fetch(:header_section), TypeDeChamp.type_champs.fetch(:explication)]
+    !types_without_label.include?(@champ.type_champ)
+  end
+
+  def component_class
+    "EditableChamp::#{@champ.type_champ.camelcase}Component".constantize
+  end
+
+  def html_options
+    {
+      class: "editable-champ-#{@champ.type_champ} #{@champ.visible? ? '' : 'hidden'}",
+      id: @champ.input_group_id,
+      data: stimulus_controller
+    }
+  end
+
+  def stimulus_controller
+    if !@champ.repetition? && !@champ.non_fillable?
+      # This is an editable champ. Lets find what controllers it might need.
+      controllers = []
+
+      # This is a public champ – it can have an autosave controller.
+      if @champ.public?
+        # This is a champ on dossier in draft or en construction state. Activate autosave.
+        if @champ.dossier.brouillon? || @champ.dossier.en_construction?
+          controllers << 'autosave'
+        end
+      end
+
+      # This is a dropdown champ. Activate special behaviours it might have.
+      if @champ.simple_drop_down_list? || @champ.linked_drop_down_list?
+        controllers << 'champ-dropdown'
+      end
+
+      if controllers.present?
+        { controller: controllers.join(' ') }
+      end
```

Comment:
il se passe quoi si tu testes pas la présence ?
```suggestion
      { controller: controllers.join(' ') }
```